specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json

```diff
@@ -0,0 +1,1981 @@
+        },
+        "managedDiskIdentity": {
+          "description": "The details of Managed Identity of Disk Encryption Set used for Managed Disk Encryption",
+          "$ref": "#/definitions/ManagedIdentityConfiguration"
+        },
+        "diskEncryptionSetId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource Id of the managed disk encryption set."
+        },
+        "encryption": {
+          "type": "object",
+          "description": "Encryption properties for databricks workspace",
+          "properties": {
+            "entities": {
+              "$ref": "#/definitions/EncryptionEntitiesDefinition",
+              "description": "Encryption entities definition for the workspace."
+            }
+          },
+          "required": [
+            "entities"
+          ]
+        },
+        "enhancedSecurityCompliance": {
+          "description": "Contains settings related to the Enhanced Security and Compliance Add-On.",
+          "$ref": "#/definitions/EnhancedSecurityComplianceDefinition"
+        },
+        "privateEndpointConnections": {
+          "readOnly": true,
+          "description": "Private endpoint connections created on the workspace",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          }
+        },
+        "publicNetworkAccess": {
+          "type": "string",
+          "description": "The network access type for accessing workspace. Set value to disabled to access workspace only via private link.",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "PublicNetworkAccess"
+          }
+        },
+        "requiredNsgRules": {
+          "type": "string",
+          "description": "Gets or sets a value indicating whether data plane (clusters) to control plane communication happen over private endpoint. Supported values are 'AllRules' and 'NoAzureDatabricksRules'. 'NoAzureServiceRules' value is for internal use only.",
+          "enum": [
+            "AllRules",
+            "NoAzureDatabricksRules",
+            "NoAzureServiceRules"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "RequiredNsgRules"
+          }
+        },
+        "defaultCatalog": {
+          "description": "Properties for Default Catalog configuration during workspace creation.",
+          "$ref": "#/definitions/DefaultCatalogProperties"
+        },
+        "isUcEnabled": {
+          "readOnly": true,
+          "description": "Indicates whether unity catalog enabled for the workspace or not.",
+          "type": "boolean"
+        },
+        "accessConnector": {
+          "description": "Access Connector Resource that is going to be associated with Databricks Workspace",
+          "type": "object",
+          "properties": {
+            "id": {
```

Comment:
The name "id" of a property should not be used in nested definitions. Please use "connectorId" or "resourceId" or similar here. Also, use [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details) here.

Reply:
Thanks for sharing this recommendation, however this feature has gone through private preview review and got approved for using id and now are publishing to public preview. Please refer the below PR ARM approved it, going forward we will keep this guidance in mind and use appropriate naming. 


https://github.com/Azure/azure-rest-api-specs-pr/pull/11313

Reply:
I agree it's unfortunate this wasn't caught sooner. Please note for future: top-level ARM resource envelope property names (name, id, type, etc.) should never be reused within the property bag. I'm leaving it to you whether to make this change right now or not: the recommendation from ARM is to make it now before your API goes public.

Reply:
While I wish to consider making the changes, but that will cause the existing customers breaking change. So I will leave this as is. Going forward, I will make sure our team is educated on this and follow the guidance.